openassessment/workflow_batch_update_api.py

```diff
@@ -0,0 +1,269 @@
+"""
+Provides functionality to batch update ORA workflows for different scopes
+"""
+
+import logging
+import time
+import datetime
+from django.utils import timezone
+
+from opaque_keys.edx.keys import UsageKey
+from openassessment.runtime_imports.functions import modulestore
+from openassessment.assessment.models import PeerWorkflow
+from openassessment.workflow import api
+
+logger = logging.getLogger(__name__)
+
+
+def update_workflows_for_ora_block(item_id):
+    """
+    Updates ORA workflows created for the given ORA Block
+
+    Args:
+    item_id (str): Identifier for the ORA Block
+        e.g. 'block-v1:edX+DemoX+Demo_Course+type@openassessment+block@1676f4b05f0642249ff724e7c07d869e'
+
+    Raises:
+        OraWorkflowBatchUpdateException: If batch process fails and cannot continue, e.g. number of errors
+                                        threshold was exceeded, etc.
+    """
+    try:
+        start = time.time()
```

Comment:
Proposed refactor: replace manual time calculation with a timing decorator, can remove those 7 lines of code from each function.

Reply:
Makes sense for simple cases, however we log other information besides args and execution times, so keeping it in function code.

Reply:
I have made changes, however, these are my comments:

Decorators (or annotations in Java) quite often (and in case of the single API boundaries often to very often) affect code readability. I would rather have more verbose imperatively written code and see everything that's going on, than additional complexity that is scattered all over the place and required additional hops to understand logic flow. 

E.g. in this case I had to introduce additional return object wrapper, with not clean (**kwargs antipattern) constructor implementation and the decorator function itself. As a result we have slightly less verbose code at the expense of complexity and readability. I would say the net result is 0